### PR TITLE
Fix: Put back the prognostic run diags step in the train-diags-prog template

### PR DESCRIPTION
**Pull request by AnnaKwa**
The  multiple model template for train-diags-prog workflows didn't have the report step, so the PR that made it the only train-diags-prog template (https://github.com/VulcanClimateModeling/vcm-workflow-control/pull/88) accidentally removed this step from the end to end workflow.

Added public API:
- optional param `prognostic-report-flags` in the `train-diags-prog` template, use to set the verification dataset for the report step.

Resolves #<https://github.com/VulcanClimateModeling/fv3net/issues/1095>

### CODE REVIEW

File: workflows/argo/train-diags-prog.yaml [hunk 1]
```diff
@@ -110,6 +111,19 @@ spec:
...
+          name: prognostic-run-diags
+          template: diagnostics
+        arguments:
+          parameters:
+              - name: run
+                value: "{{inputs.parameters.root}}/prognostic_run"
+              - name: output
+                value: "{{inputs.parameters.root}}/prognostic_run_diags"
+              - name: flags
+                value: "{{inputs.parameters.prognostic-report-flags}}"
```
Comment (oliverwm1) [inline workflows/argo/train-diags-prog.yaml:126]:
```suggestion
                value: "{{inputs.parameters.online-diags-flags}}"
```

**Review by oliverwm1 — COMMENTED** [review]
Thanks! I have a name suggestion, let me know what you think.

Also could you make sure the README in `workflows/argo` is up to date for this workflow?

File: workflows/argo/README.md [hunk 2]
```diff
@@ -248,6 +248,11 @@ formatted as `[{name: model_name, config: model_config}, ...]`, and where the in
 as for the `training` workflow.  In practice it may be easiest to write this as a YAML file and then converted to
 JSON format using `yq . config.yml` in a submission script.
 
+The default behavior for the final `prognostic-run-diags.diagnostics-step` is to use the default verifcation dataset
```
Comment (oliverwm1) [inline workflows/argo/README.md:251]:
```suggestion
The default behavior for the final `prognostic-run-diags.diagnostics-step` is to use the default verification dataset
```

File: workflows/argo/README.md [hunk 3]
```diff
@@ -248,6 +248,11 @@ formatted as `[{name: model_name, config: model_config}, ...]`, and where the in
...
+(`40day_may2020`) to calculate the metrics. If this is not the appropriate verification data to use, make sure to specifiy
```
Comment (oliverwm1) [inline workflows/argo/README.md:252]:
```suggestion
(`40day_may2020`) to calculate the metrics. If this is not the appropriate verification data to use, make sure to specify
```

**Review by oliverwm1 — APPROVED** [review]
LGMT!

**Comment by AnnaKwa** [comment]
Note- this did pass the E2E integration test, that pass was marked as a build fail due to a broken path in the dataflow test (fixed when master was merged in).

**Pull request merged**